src/mz/src/server.rs

```diff
@@ -0,0 +1,50 @@
+use std::net::{Ipv4Addr, SocketAddr, SocketAddrV4};
+
+use axum::{
+    extract::Query,
+    response::IntoResponse,
+    routing::{get, IntoMakeService},
+    Router, Server,
+};
+use mz_frontegg_auth::AppPassword;
+use reqwest::StatusCode;
+use serde::Deserialize;
+use tokio::sync::mpsc::UnboundedSender;
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct BrowserAPIToken {
+    client_id: String,
+    secret: String,
+}
+
+/// Request handler for the server waiting the browser API token creation
+// Axum requires the handler be async even though we don't await
+#[allow(clippy::unused_async)]
+async fn request(
+    Query(BrowserAPIToken { secret, client_id }): Query<BrowserAPIToken>,
+    tx: UnboundedSender<AppPassword>,
+) -> impl IntoResponse {
+    tx.send(AppPassword {
+        client_id: client_id.parse().unwrap(),
+        secret_key: secret.parse().unwrap(),
+    })
```

Comment:
If the user clicks `no` on the web console to create an app password then the unwrap on line 29 is triggered. We should handle that case more gracefully.

Reply:
It also hangs forever if you panic in this task. If anything fails you'll need to explicitly exit.

Reply:
I've added multiple error handlers for the App Password. The only one missing is the transaction panicking, which has a low chance to occur but anyways we should have something there.